mmpose/engine/hooks/ema_hook.py

```diff
@@ -0,0 +1,65 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import math
+from typing import Optional
+
+import torch
+import torch.nn as nn
+from mmengine.model import ExponentialMovingAverage
+from torch import Tensor
+
+from mmpose.registry import MODELS
+
+
+@MODELS.register_module()
+class ExpMomentumEMA(ExponentialMovingAverage):
+    """Exponential moving average (EMA) with exponential momentum strategy,
```

Comment:
Please add paper reference and attribution information if the code is ported from other projects.